code/scheme/guile-r7rs/bbhw.scm

```diff
@@ -0,0 +1,27 @@
+(import (scheme base)
+        (only (scheme process-context) command-line)
+        (only (srfi 18) current-time time->seconds seconds->time thread-sleep!))
+
+(define count
+  (let get-count ((input (cdr (command-line))))
+    (cond ((and (integer? input) (>= input 0)) input)
+          ((and (string? input) (string->number input)) => get-count)
+          ((pair? input) (get-count (car input)))
+          ((null? input) (get-count #f))
+          ((and (string? input) (string=? "" input)) (get-count #f))
+          (else
+           (if input
+               (begin
+                 (write-string "Invalid countdown ")
+                 (display input)
+                 (write-string ", try again...")
+                 (newline)))
+           (write-string "countdown: ")
+           (get-count (read-line))))))
+
+(write-string "World, Hello...") (flush-output-port)
+(do ((i count (- i 1)))
+    ((zero? i))
+  (display i) (write-string "...") (flush-output-port)
+  (thread-sleep! (seconds->time (+ (time->seconds (current-time)) 1))))
```

Comment:
(thread-sleep! (seconds->time (+ (time->seconds (current-time)) 1)))

This is a complicated detour for `(thread-sleep! 1)`. 

From SRFI 18: "an exact or inexact real number represents a relative time in seconds from the moment the primitive was called"

Reply:
True, missed that, thanks! Simplified in d362dbd.